### PR TITLE
e2e refactor

### PR DESCRIPTION
**Pull request by pmikolajczyk41**
- moved auxiliary functions to a dedicated file (to avoid junked `main.rs`)
- moved waiting utilities to a dedicated file (complex logic outside `main.rs`)
- reporting about test status (to see the progress)
- reduce unit-creation-time from 500ms to 50ms (fixing #195)

### CODE REVIEW

File: e2e-tests/src/utils.rs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        "transfer",
+        GenericAddress::Id(target.clone()),
+        Compact(value)
+    );
+
+    let tx_hash = connection
+        .send_extrinsic(tx.hex_encode(), XtStatus::Finalized)
+        .unwrap()
+        .expect("Could not get tx hash");
+    info!("[+] Transaction hash: {}", tx_hash);
```
Comment (fbielejec) [inline e2e-tests/src/utils.rs:86]:
```suggestion
    info!("[+] Transfer transaction hash: {}", tx_hash);
```

**Review by fbielejec — COMMENTED** [review]
minor comment, else LGTM

File: e2e-tests/src/utils.rs (hunk 1 above)
Comment (pmikolajczyk41) [reply e2e-tests/src/utils.rs]: done

**Review by kostekIV — APPROVED** [review]

**Review by fbielejec — APPROVED** [review]

File: e2e-tests/src/main.rs [hunk 2]
```diff
@@ -34,18 +26,29 @@ fn main() -> anyhow::Result<()> {
...
+fn run<T>(
+    testcase: fn(Config) -> anyhow::Result<T>,
+    name: &str,
+    config: Config,
+) -> anyhow::Result<()> {
+    println!("Running test: {}", name);
+    let start = Instant::now();
+    testcase(config).map(|_| {
+        let elapsed = Instant::now().duration_since(start);
+        println!("OK! ({:?})", elapsed)
```
[the remark this answers is not in the document]
Comment (pmikolajczyk41) [reply e2e-tests/src/main.rs]: done

**Pull request merged**